client/src/modules/reports/generate/stock_exit/stock_exit.html

```diff
@@ -0,0 +1,90 @@
+<bh-report-preview
+  ng-if="ReportConfigCtrl.previewGenerated"
+  source-document="ReportConfigCtrl.previewResult"
+  on-clear-callback="ReportConfigCtrl.clearPreview()"
+  on-save-callback="ReportConfigCtrl.requestSaveAs()">
+</bh-report-preview>
+
+<div ng-show="!ReportConfigCtrl.previewGenerated">
+  <div class="row">
+    <div class="col-md-12">
+      <h3 translate>REPORT.STOCK.EXIT_REPORT</h3>
+      <p class="text-info" translate>REPORT.STOCK.EXIT_REPORT_DESCRIPTION</p>
+    </div>
+  </div>
+
+  <div class="row" style="margin-top : 10px">
+    <div class="col-md-6">
+      <div class="panel panel-default">
+        <div class="panel-heading">
+          <span translate>REPORT.UTIL.OPTIONS</span>
+        </div>
+
+        <div class="panel-body">
+
+        <form name="ConfigForm" bh-submit="ReportConfigCtrl.preview(ConfigForm)" novalidate autocomplete="off">
+
+          <!-- select depot -->
+          <bh-depot-select
+            depot-uuid="ReportConfigCtrl.depot.uuid"
+            on-select-callback="ReportConfigCtrl.onSelectDepot(depot)"
+            required="true"
+            validate-trigger="ConfigForm.$submitted">
+            <bh-clear on-clear="ReportConfigCtrl.clear('depot')"></bh-clear>
```

Comment:
Do you need this `clear()` method?  To me, if you cannot submit this report without a depot then there is no need for supporting clearing it.